open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md

```diff
@@ -14,3 +14,250 @@ kernelspec:
+    ingredient_df = ingredient_df.sort_values(by='value', ascending=False,
+    inplace=False)
+    return ingredient_df
+```
+
+Now you can use that function to get an idea of top ten most popular ingredients by cuisine.
+
+2 . Call `create_ingredient()` and plot it calling `barh()`:
+
+```{code-cell}
+thai_ingredient_df = create_ingredient_df(thai_df)
+thai_ingredient_df.head(10).plot.barh()
+```
+
+3 . Do the same for the japanese data:
+
+```{code-cell}
+japanese_ingredient_df = create_ingredient_df(japanese_df)
+japanese_ingredient_df.head(10).plot.barh()
+```
+
+4 . Now for the chinese ingredients:
+
+```{code-cell}
+chinese_ingredient_df = create_ingredient_df(chinese_df)
+chinese_ingredient_df.head(10).plot.barh()
+```
+
+5 . Plot the indian ingredients:
+
+```{code-cell}
+indian_ingredient_df = create_ingredient_df(indian_df)
+indian_ingredient_df.head(10).plot.barh()
+```
+
+6 . Finally, plot the korean ingredients:
+
+```{code-cell}
+korean_ingredient_df = create_ingredient_df(korean_df)
+korean_ingredient_df.head(10).plot.barh()
+```
+
+7 . Now, drop the most common ingredients that create confusion between distinct cuisines, by calling `drop()`:
+
+Everyone loves rice, garlic and ginger!
+
+```{code-cell}
+:tags: [output_scroll]
+
+feature_df= df.drop(['cuisine','Unnamed: 0','rice','garlic','ginger'], axis=1)
+labels_df = df.cuisine #.unique()
+feature_df.head()
+```
+
+## Balance the dataset
+
+Now that you have cleaned the data, use [SMOTE](https://imbalanced-learn.org/dev/references/generated/imblearn.over_sampling.SMOTE.html) - "Synthetic Minority Over-sampling Technique" - to balance it.
+
+1 . Call `fit_resample()`, this strategy generates new samples by interpolation.
+
+```{code-cell}
+oversample = SMOTE()
+transformed_feature_df, transformed_label_df = oversample.fit_resample(feature_df, labels_df)
+```
+
+By balancing your data, you'll have better results when classifying it. Think about a binary classification. If most of your data is one class, a ML model is going to predict that class more frequently, just because there is more data for it. Balancing the data takes any skewed data and helps remove this imbalance. 
+
+2 . Now you can check the numbers of labels per ingredient:
+
+```{code-cell}
+print(f'new label count: {transformed_label_df.value_counts()}')
+print(f'old label count: {df.cuisine.value_counts()}')
+```
+
+The data is nice and clean, balanced, and very delicious!
+
+3 . The last step is to save your balanced data, including labels and features, into a new dataframe that can be exported into a file:
+
+```{code-cell}
+transformed_df = pd.concat([transformed_label_df,transformed_feature_df],axis=1, join='outer')
+```
+
+4 . You can take one more look at the data using `transformed_df.head()` and `transformed_df.info()`. Save a copy of this data for use in future sections:
+
+```{code-cell}
+transformed_df.head()
+transformed_df.info()
+transformed_df.to_csv("../../assets/data/cleaned_cuisines.csv")
```

Comment:
You don't need to have this file in git. Please add it to the .gitignore.

Reply:
I don't understand, how to do it? what is .gitignore?

Reply:
which file should be added to the .gitignore.?  what is the .gitignore. use for ?

Reply:
../../assets/data/cleaned_cuisines.csv -> ./cleaned_cuisines.csv

Then add `open-machine-learning-jupyter-book/ml-fundamentals/classification/cleaned_cuisines.csv` to gitignore

Reply:
The goal is to exclude this dynamically generated file from the git diff.